public/pages/workflow_detail/workflow_detail.test.tsx

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import { render } from '@testing-library/react';
+import { Provider } from 'react-redux';
+
+import {
+  BrowserRouter as Router,
+  RouteComponentProps,
+  Route,
+  Switch,
+} from 'react-router-dom';
+import { store } from '../../store';
+import { WorkflowDetail } from './workflow_detail';
+import { WorkflowDetailRouterProps } from '../../pages';
+
+jest.mock('../../services', () => {
+  const { mockCoreServices } = require('../../../test');
+  return {
+    ...jest.requireActual('../../services'),
+    ...mockCoreServices,
+  };
+});
+
+const renderWithRouter = () =>
+  render(
+    <Provider store={store}>
+      <Router>
+        <Switch>
+          <Route
+            path="/workflow/:workflowId"
+            render={(props: RouteComponentProps<WorkflowDetailRouterProps>) => (
+              <WorkflowDetail setActionMenu={jest.fn()} {...props} />
+            )}
+          />
+        </Switch>
+      </Router>
+    </Provider>
+  );
+
+describe('WorkflowDetail', () => {
+  test('renders the page', () => {
+    const { container, baseElement } = renderWithRouter();
+    expect(baseElement.tagName).toBe('BODY');
+    expect(container.tagName).toBe('DIV');
```

Comment:
This isn't a useful thing to test. We should test that certain wording exists, or other descriptors of components that we would be expecting when this renders. I would recommend looking at the AD OSD plugin unit tests for more examples. You can also refer to the [testing-library documentation](https://testing-library.com/docs/queries/about/) for more info and examples.

At a high-level, we use `jest` as the test framework, and `@testing-library/react` as a helper library for rendering and pulling out UI elements. A basic flow for testing UI components/pages is:
1. Render the component using `render()` (optionally with different props for different states)
2. Run `testing-library` queries (linked above) against that returned component to ensure certain elements exist etc.
3. Assert that those queries return what is expected using `jest`, such as `expect()`